var/spack/repos/builtin/packages/libf77zmq/package.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from spack import *
+
+
+class Libf77zmq(MakefilePackage):
+    """Fortran binding for the ZeroMQ communication library"""
+
+    homepage = "http://zguide.zeromq.org/"
+    url      = "https://github.com/zeromq/f77_zmq/archive/4.3.1.tar.gz"
+
+    maintainers = ['scemama']
+
+    version('4.3.1', sha256='a15d72d93022d3e095528d2808c7767cece974a2dc0e2dd95e4c122f60fcf0a8')
+
+    depends_on('libzmq')
+    depends_on('python', type='build')
+
+    def setup_build_environment(self, env):
+        env.append_flags('CFLAGS', '-O3')
+        env.append_flags('CFLAGS', '-g')
+
+    def edit(self, spec, prefix):
+        cflags = os.environ.get('CFLAGS')
+        makefile = FileFilter('Makefile')
+        makefile.filter('CC=.*',
+                        'CC={0} {1}'.format(self.compiler.cc,
+                                            self.compiler.pic_flag))
```

Comment:
Why not append `self.compiler.pic_flag` to `CFLAGS` instead?

Reply:
```make
libf77zmq.so: f77_zmq.o
        $(CC) -shared $^ -o $@

libf77zmq.a: f77_zmq.o 
        $(AR) cr $@  $^

f77_zmq.o: f77_zmq.c f77_zmq.h
        $(CC) $(CFLAGS) -c f77_zmq.c -o $@
```
`CFLAGS` does not appear at the link, and -fPIC should also be there.